libraries/chain/db_market.cpp

```diff
@@ -76,11 +76,8 @@ void database::globally_settle_asset( const asset_object& mia, const price& sett
    {
       if( before_core_hardfork_342 )
       {
+         //Note: If here, the call order may be paying nothing (Yes, this happened)
```

Comment:
This didn't happen.

Reply:
Hrmmm. I just ran it again, and found 8 instances on mainnet where "pays" is zero. Am I misunderstanding the code (or perhaps misunderstanding your comment)?

Reply:
Here is a list of all log messages related to #184 before the hf (produced by replaying with `3.3.1`):
TL;DR: some variant C, some variant F
```
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #17167173                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #19843236                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #22370884                 db_market.cpp:750
th_a:?unnamed?             do_apply ] Something for nothing issue (#184, variant F) occurred at block #22525298                 asset_evaluator.cpp:879
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #23592820                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #23593831                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #23676128                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #24184740                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #24310110                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #25264776                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #25319453                 db_market.cpp:750
th_a:?unnamed?             do_apply ] Something for nothing issue (#184, variant F) occurred at block #25369774                 asset_evaluator.cpp:879
th_a:?unnamed?             do_apply ] Something for nothing issue (#184, variant F) occurred at block #25727532                 asset_evaluator.cpp:879
th_a:?unnamed?             do_apply ] Something for nothing issue (#184, variant F) occurred at block #25930044                 asset_evaluator.cpp:879
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #26155800                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #26722712                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #26722721                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #26722749                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #27011422                 db_market.cpp:750
th_a:?unnamed?             do_apply ] Something for nothing issue (#184, variant F) occurred at block #27352798                 asset_evaluator.cpp:879
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #27887880                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #27887886                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #27936676                 db_market.cpp:750
```

Reply:
```
//Note: If here, the call order may be paying nothing (Yes, this happened)
pays = call_itr->get_debt() * settlement_price; // round down, in favor of call order
if ( pays.amount == 0 )
{
     wlog("Before HF 342, paying nothing for collateral ${asset} at block ${blk}", 
            ("asset", call_itr->get_collateral())
            ("blk", head_block_num())
       );
}
```
results in:
```
default.log:2019-09-10T21:42:50        th_a:?unnamed? globally_settle_asse ] Before HF 342, paying nothing for collateral {"amount":30000000,"asset_id":"1.3.0"} at block 14032429			db_market.cpp:86
default.log:2019-09-10T21:42:50        th_a:?unnamed? globally_settle_asse ] Before HF 342, paying nothing for collateral {"amount":"2264664694","asset_id":"1.3.0"} at block 14032429			db_market.cpp:86
default.log:2019-09-10T21:42:50        th_a:?unnamed? globally_settle_asse ] Before HF 342, paying nothing for collateral {"amount":149600000,"asset_id":"1.3.0"} at block 14032429			db_market.cpp:86
default.log:2019-09-10T21:42:58        th_a:?unnamed? globally_settle_asse ] Before HF 342, paying nothing for collateral {"amount":2000000000,"asset_id":"1.3.0"} at block 14689927			db_market.cpp:86
default.log:2019-09-10T21:42:58        th_a:?unnamed? globally_settle_asse ] Before HF 342, paying nothing for collateral {"amount":290000000,"asset_id":"1.3.0"} at block 14689927			db_market.cpp:86
default.log:2019-09-10T21:42:58        th_a:?unnamed? globally_settle_asse ] Before HF 342, paying nothing for collateral {"amount":100000000,"asset_id":"1.3.0"} at block 14689927			db_market.cpp:86
default.log:2019-09-10T21:42:58        th_a:?unnamed? globally_settle_asse ] Before HF 342, paying nothing for collateral {"amount":935875108,"asset_id":"1.3.0"} at block 14689927			db_market.cpp:86
default.log:2019-09-10T21:42:58        th_a:?unnamed? globally_settle_asse ] Before HF 342, paying nothing for collateral {"amount":1165000000,"asset_id":"1.3.0"} at block 14689927			db_market.cpp:86
```

Reply:
Prediction markets which have `0` as outcome can pay nothing for the collateral. That's why there was a check for PM there.

Reply:
So, I guess there is no need for a comment here, as this is normal. I will rearrange.